airflow/serialization/serialized_objects.py

```diff
@@ -1086,8 +1087,15 @@ def deserialize_task_group(
             for key in ["prefix_group_id", "tooltip", "ui_color", "ui_fgcolor"]
         }
         group = SerializedTaskGroup(group_id=group_id, parent_group=parent_group, **kwargs)
+
+        ref = weakref.proxy(group)
+
+        def set_ref(task: BaseOperator) -> BaseOperator:
+            task.task_group = ref
+            return task
```

Comment:
```suggestion
        def set_ref(task: BaseOperator) -> BaseOperator:
            task.task_group = weakref.proxy(group)
            return task
```

Reply:
Is there a problem re-using the proxy?

Reply:
Not really; the two are actually equivalent because proxy is implemented to return the same thing from one object.

```pycon
>>> import weakref
>>> class A: pass
...
>>> a = A()
>>> id(weakref.proxy(a))
139967946164800
>>> id(weakref.proxy(a))
139967946164800
```